distributed/shuffle/_scheduler_extension.py

```diff
@@ -122,6 +124,14 @@ async def barrier(self, id: ShuffleId, run_id: int) -> None:
             msg=msg, workers=list(shuffle.participating_workers)
         )
 
+    async def restrict_task(
+        self, id: ShuffleId, run_id: int, key: str, worker: str
+    ) -> None:
+        shuffle = self.states[id]
+        if shuffle.run_id != run_id:
+            raise RuntimeError()
```

Comment:
I think we should also check if the shuffle this run_id points to is still valid. We don't want to have stale requests modify the restrictions.

Reply:
We should probably document this somewhere, but if it's in `ShuffleSchedulerExtension.states`, it's an active and valid shuffle instance. Otherwise, it would have been dropped from there.